src/compliment/utils.clj

```diff
@@ -181,25 +185,45 @@ Note that should always have the same value, regardless of OS."
       (->> (for [^String file (all-files-on-classpath classpath)
                  :when (and (.endsWith file ".class") (not (.contains file "__"))
                             (not (.contains file "$")))]
-             (.. (if (.startsWith file File/separator)
-                   (.substring file 1) file)
+             (.. (ensure-no-leading-slash file)
                  (replace ".class" "")
                  ;; Address the issue #79 , on Windows, for prefix such
                  ;; as "java.util.", the list of candidates was empty.
                  (replace resource-separator ".")))
            (group-by #(subs % 0 (max (.indexOf ^String % ".") 0)))))))
 
-(defn namespaces-on-classpath
-  "Returns the list of all Clojure namespaces obtained by classpath scanning."
-  []
-  (let [classpath (classpath)]
-    (cache-last-result ::namespaces-on-classpath classpath
-      (set (for [^String file (all-files-on-classpath classpath)
-                 :when (and (.endsWith file ".clj")
-                            (not (.startsWith file "META-INF")))
-                 :let [[_ ^String nsname] (re-matches #"[^\w]?(.+)\.clj" file)]
-                 :when nsname]
-             (.. nsname (replace resource-separator ".") (replace "_" "-")))))))
+(defn namespaces&files-on-classpath
+  "Returns the list of all clj(c) namespaces obtained by classpath scanning.
+
+  Options:
+  - `extensions` (default `#{\"clj\", \"cljc\"}`) - what files to consider. Should be subset of `#{\"clj\", \"cljc\", \"cljs\"}`."
+  ([] (namespaces&files-on-classpath nil))
+  ([{:keys [extensions] :or {extensions #{"clj" "cljc"}}}]
+   (let [classpath  (classpath)
+         extensions (set extensions)
+         nses
+         (cache-last-result ::namespaces-on-classpath classpath
+           (for [^String file (all-files-on-classpath classpath)
+                 :when        (and (or (.endsWith file ".clj")
+                                       (.endsWith file ".cljc")
+                                       (.endsWith file ".cljs"))
+                                   (not (.startsWith file "META-INF")))
+                 :let         [file (ensure-no-leading-slash file)
+                               [_ ^String nsname ^String extension] (re-matches #"[^\w]?(.+)\.(clj[sc]?)" file)]
+                 :when        nsname]
+             (let [ns-str (.. nsname (replace resource-separator ".") (replace "_" "-"))]
+               {:ns ns-str, :file file, :extension extension})))
+         xf         (comp (filter (comp extensions :extension))
+                          (map #(dissoc % :extension)))]
+     (into #{}
+           (transduce xf conj nses)))))
+
+(defn ^:deprecated namespaces-on-classpath []
+  (into #{}
+        (transduce
+         (map :ns)
+         conj
+         (namespaces&files-on-classpath {:extensions #{"clj" "cljc"}}))))
```

Comment:
You don't need both `into` and `transduce`. It's either `(into #{} (map :ns) coll)` or `(transduce (map :ns) conj #{} coll)`.